pkg/pillar/nireconciler/linux_config.go

```diff
@@ -636,6 +636,40 @@ func (r *LinuxNIReconciler) getIntendedNIL3Cfg(niID uuid.UUID) dg.Graph {
 			}, nil)
 		}
 	}
+	// Add statically defined routes into the NI routing table.
+	// No need to do this if GW is one of the apps and traffic from one app
+	// to another is just forwarded by the host, not routed.
+	for _, route := range ni.config.StaticRoutes {
+		isAppGW := r.getNISubnet(ni).Contains(route.Gateway)
+		if isAppGW && r.disableAllOnesNetmask {
+			// Route is not needed inside the host, traffic is just forwarded
+			// by the bridge.
+			continue
+		}
+		if !isAppGW && !r.routeGwIsConnected(route, ni.bridge.Uplink) {
```

Comment:
I guess there is an unstated assumption about the routes only being applicable to a particular NI, which I think we must document.

If I have an app instance connected to two NIs; the default local NI and a switch NI out eth1, then I might assume I can use EVE to publish a route on the first where the gateway is on the seconnd NI. AFAIK a guest VM like Linux would accept such a route (unless there is something special in some dhcp client code to tie to received routes to the interface on which the dhcp client received the option.)

Reply:
Yes, routes configured for a given NI should only apply to that network instance and interfaces connected to it. I will make sure to document this.

I believe that with this restriction it leads to cleaner configurations. In your example, it should be the external DHCP server used by the switch network instance where such route is configured.

Reply:
How much documentation are we planning for this feature?

Reply:
Not much, I will prepare some paragraph for eve docs once we finish this review and clarify all the edge cases.